### PR TITLE
Autoinstall extension in chrome

### LINKED ISSUE #6076: Install chrome extension automatically
It would be nice if the Chrome extension would be automatically installed during the main installation routine. 

Shouldn't be that hard to implement: https://developer.chrome.com/extensions/external_extensions
@LyzardKing interested? 😄 

(For Firefox something like this is sadly not supported: https://blog.mozilla.org/addons/2019/10/31/firefox-to-discontinue-sideloaded-extensions/)

### PR DESCRIPTION
**Pull request by LyzardKing**
This fixes #6076 
Draft PR to build macos releases with the extension json files.
This should add the postinst/prerm scripts for deb files to add the extension automatically
Add initial json files on macos

### CODE REVIEW

**Comment by LyzardKing** [comment]
@tobiasdiez 
The linux (deb) install works.. but only (I think) if the extension isn't already been installed and removed..I need to verify this..
The rpm is setup, but needs testing to verify it works..
For the snap we would need another permission (that would probably not be granted, but we can ask..
The macos part is setup, but I need someone to test it.

**Comment by Siedlerchr** [comment]
I have retriggered a build, because mac was not building For the mac os  it's best to ask in the gitter channel. the version will be avaiable under
https://builds.jabref.org/pull/6442/merge/

File: buildres/linux/postinst [hunk 1]
```diff
@@ -19,10 +19,14 @@ set -e
...
     configure)
+        # Install the native-messaging host script for firefox/chrome/chromium
         install -D -m0755 /opt/jabref/lib/native-messaging-host/firefox/org.jabref.jabref.json /usr/lib/mozilla/native-messaging-hosts/org.jabref.jabref.json
         install -D -m0755 /opt/jabref/lib/native-messaging-host/chromium/org.jabref.jabref.json /etc/chromium/native-messaging-hosts/org.jabref.jabref.json
         install -D -m0755 /opt/jabref/lib/native-messaging-host/chromium/org.jabref.jabref.json /etc/opt/chrome/native-messaging-hosts/org.jabref.jabref.json
-        DESKTOP_COMMANDS_INSTALL
+        # Trigger an auto-install of the browser addon for chrome/chromium browsers
+        install -D -m0644 /opt/jabref/lib/native-messaging-host/chromium/bifehkofibaamoeaopjglfkddgkijdlh.json /opt/google/chrome/extensions/bifehkofibaamoeaopjglfkddgkijdlh.json
+        install -D -m0644 /opt/jabref/lib/native-messaging-host/chromium/bifehkofibaamoeaopjglfkddgkijdlh.json /usr/share/google-chrome/extensions/bifehkofibaamoeaopjglfkddgkijdlh.json
+	DESKTOP_COMMANDS_INSTALL
```
Comment (tobiasdiez) [inline buildres/linux/postinst:29]: Minor: indent is off here

Comment (LyzardKing) [reply buildres/linux/postinst:29]: ah.. thanks. I guess the IDE changed my settings.

**Comment by LyzardKing** [comment]
@tobiasdiez I tried retriggering the build..
on macos the build succeeds, but it then stops and does not build the installer..

**Comment by tobiasdiez** [comment]
Yeah, not sure where the problem is. Your changes look fine to me. I tried to rename the postinstall file to see if that's the problem.

For reference: https://docs.oracle.com/en/java/javase/14/jpackage/override-jpackage-resources.html#GUID-405708DC-0243-49FC-84D9-B2A7F0A011A9
(could be that the postinstall is PKG specific and we also build a dmg)

**Comment by Siedlerchr** [comment]
A while ago I enabled verbose output for the jlink/jpackage task on ci, so that we see what the task does.

**Comment by LyzardKing** [comment]
@tobiasdiez the postinstall is only used in the pkg.. but as far as I can tell the dmg should just ignore it

**Comment by tobiasdiez** [comment]
Strange...it works with the postinstall file renamed :( Any idea what to do? Can we exclude the postinstall file for the dmg generation?

**Comment by LyzardKing** [comment]
We could try and rename the file to the correct name before setting the .pkg
So  the .app is without...
But I'm not sure, since the .app is not the partthat is failing here..

**Comment by Siedlerchr** [comment]
I merged the master and renamed it back, lets see if it works now

**Comment by Siedlerchr** [comment]
screw it, the damn apache repo is offline or has issues

**Comment by Siedlerchr** [comment]
I triggered the build and it seems to have build now @LyzardKing

**Comment by LyzardKing** [comment]
@Siedlerchr Is it building with the proper filename?

**Comment by Siedlerchr** [comment]
Yes, I renamed it back to "postinstall" without extension

**Comment by LyzardKing** [comment]
Great! I asked in the gitter channel if someone with a mac can test it out!
Hopefully it should work

**Comment by level1807** [comment]
So is the .pkg supposed to auto-install the Chrome extension now? It's not doing it for me.

**Comment by LyzardKing** [comment]
@level1807 That would be the plan.
Unfortunately I can't figure out what is missing...
I have another couple of things to test later this week..

**Comment by tobiasdiez** [comment]
@LyzardKing What's the status here? We are planning to release 5.1 soon and it would be nice to include this PR. If Mac has problems, then we can leave it for now.

**Comment by LyzardKing** [comment]
The linux version (non snap) works.
I'm still debugging the macos version, since it works only sometimes. It's very flaky.
In a few days I should have access to a macbook so I can test it properly.
I also need to check the windows integration...
When is the release scheduled?

**Comment by tobiasdiez** [comment]
That sounds very good! We were hoping to release in about two weeks.

**Comment by LyzardKing** [comment]
It should be enough time to attempt at fixing the macos build.
I'll keep you posted on my progress

File: buildres/mac/jabrefHost.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python3
```
Comment (tobiasdiez) [inline buildres/mac/jabrefHost.py:1]: It is not possible to reuse the linux version?

**Comment by tobiasdiez** [comment]
We are planning on releasing this week. Do you think to find the time until then to finish this PR?

**Comment by LyzardKing** [comment]
The issues with macos seem to be that
a) there is no way to remove on uninstall the files copied on install (there is no postremove script)
b) the install works, but not consistently on different machines/vms
I intend to look at it in the next few days.. If I can get a working version on macos as well good, otherwise we can release the linux version for now..

File: buildres/mac/jabrefHost.py (hunk 2 above)
Comment (LyzardKing) [reply buildres/mac/jabrefHost.py:1]: It is basically the same file. The issue is that with the --resource-dir option in jpackage I cannot add a single file from another directory.
So I copied it and changed the path settings.

**Comment by tobiasdiez** [comment]
Ok perfect! Even if it's not working across all mac os systems, its already a huge improvement (since right now it's not working at all on mac). Maybe from the reports of people we can deduce when and when not its working.

@koppor feel free to merge this, it's good to go from my point of view. Further improvements by @LyzardKing can be included later as well, I would say.

**Comment by LyzardKing** [comment]
I'd wait a moment, since I just uploaded a change I need to test.
I'll tests it then revert the commit if it doesn't solve the issue.
I'll let you know after the builds are uploaded (and can be tested)

**Comment by Siedlerchr** [comment]
@LyzardKing  If you need someone to test on a real mac, I can help now. I just a macbook at hand

**Comment by LyzardKing** [comment]
@Siedlerchr Thanks!
I'll have (briefly) a mac tomorrow, but if you can test it it would be wonderful!

**Comment by LyzardKing** [comment]
@Siedlerchr 
it you want to test this you can find the build at https://builds.jabref.org/pull/6442/merge/

**Comment by Siedlerchr** [comment]
I don't see any files or folders or stuff in LIbrary Application Support. I tested the pkg. 
10.15.6 (19G73)

Have both firefox and chrome installed. (I'm new to mac, so just tell me where you look),
I also merged in master now, this reduces the package sizes a lot!

**Comment by koppor** [comment]
This fixes https://github.com/JabRef/jabref/issues/6076 right?

**Comment by LyzardKing** [comment]
@koppor Yes.
It however works fine on linux (at least the deb/rpm)
On macos it woks in certain configurations: I got it to work if I installed chrome and then jabref without starting chrome.
I tried to read up on the postinstall function on macos and it seems to be a bit "opinionated".
Also the `install` command works differently on macos that linux, so I switched to using manually mkdir and cp, as suggested on an old mac forum...
but I cannot get it to work properly.

**Comment by Siedlerchr** [comment]
@LyzardKing  MacOSX is deprecating bash etc and some things may not be available as default anylonger : https://github.com/scriptingosx/pkgcheck So that might explain why it's not working always.
I'm happy to test any further versions

**Comment by LyzardKing** [comment]
@Siedlerchr Do you mind running the script directly? It needs root access (sudo) but it should work.
The script tries to run the sh environment, not bash

**Comment by LyzardKing** [comment]
This might be the issue...
> So far, in macOS (including Catalina),sh invokes bash in sh-compatibility mode

**Comment by LyzardKing** [comment]
I tried forcing the proper path for /bin/sh.
@Siedlerchr try testing this version once it builds.
If it does I'd like to revert to using install to properly set the file permissions

**Comment by Siedlerchr** [comment]
@LyzardKing  The pkg file worked fine now! Both files copied to Chrome and Firefox folder.

**Comment by koppor** [comment]
Thus, this is good to go or do we need some more cleanup / testing?

**Comment by LyzardKing** [comment]
@koppor I'd like to test it with the install command, so we can explicitly set the file permissions.
@Siedlerchr Can you test it one more time after I upload the change?

**Comment by LyzardKing** [comment]
If this last version works we can merge. Windows is still missing, but we'll work on that later.
If it fails let's revert to cp again

**Comment by Siedlerchr** [comment]
Just tested it, the pgk works fine as well.

**Comment by LyzardKing** [comment]
Perfect! Thanks for the suggestion to set sh directly..
This can now be merged, then the windows version can be added later.

**Comment by Siedlerchr** [comment]
Thanks again for your contribution! This is very cool now

**Pull request merged**